src/net/sourceforge/kolmafia/request/DrinkItemRequest.java

```diff
@@ -189,7 +189,9 @@ public static final int maximumUses(
       allowOverDrink = true;
     }
 
-    int maxNumber = inebrietyLeft / inebriety + shotglass;
+    int temp = inebriety + shotglass;
+    // Don't overflow if overdrink is allowed
+    int maxNumber = temp == 0 ? Integer.MAX_VALUE - 1 : inebrietyLeft / temp;
```

Comment:
I don't think this is right.

Suppose a drink is 1 inebriety, you have 15 drunk left, and you have shotglass. You should be able to drink 16.

This calculation puts `temp` at 2, and answers 7.

Reply:
It definitely does not agree with how math works. :)
```
int maxNumber = A / B + C;
```
equals
```
int maxNumber = (A / B) + C;
```
not
```
int maxNumber = A / (B + C);

Reply:
Yeah.  You're right.  I missed that.  Fixing.